dask/dataframe/hashing.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pandas as pd
+
+from pandas.core.categorical import is_categorical_dtype
+
+
+def hash_pandas_object(obj):
+    if isinstance(obj, (pd.Series, pd.Index)):
+        h = hash_array(obj.values).astype('uint64')
+    elif isinstance(obj, pd.DataFrame):
+        cols = obj.iteritems()
+        first_series = next(cols)[1]
+        h = hash_array(first_series.values).astype('uint64')
```

Comment:
Is this `astype` needed? It appears that `hash_array` always returns a `u8`. If it is, would prefer this moved to `hash_array`, either by fixing the math so it is, or adding the `astype('u8', copy=False)` to the output there instead.